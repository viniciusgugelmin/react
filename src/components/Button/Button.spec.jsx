import { Button } from './index';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('<Button />', () => {
  it('should render the button with the text "Load more"', () => {
    expect.assertions(1);

    render(<Button text="Load more" />);

    const button = screen.getByRole('button', { name: /^load more$/i });
    expect(button).toBeInTheDocument();
  });

  it('should call function on button click', () => {
    expect.assertions(1);

    const fn = jest.fn();
    render(<Button text="Load more" fn={fn} />);

    const button = screen.getByRole('button', { name: /^load more$/i });
    userEvent.click(button);
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('should be disabled when disabled is true', () => {
    expect.assertions(1);

    render(<Button text="Load more" disabled={true} />);

    const button = screen.getByRole('button', { name: /^load more$/i });
    expect(button).toBeDisabled();
  });

  it('should be enabled when disabled is false', () => {
    expect.assertions(1);

    render(<Button text="Load more" disabled={false} />);

    const button = screen.getByRole('button', { name: /^load more$/i });
    expect(button).not.toBeDisabled();
  });
});
