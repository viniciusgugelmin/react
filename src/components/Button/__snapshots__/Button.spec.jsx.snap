// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapchot 1`] = `
<button
  class="button"
>
  Load more
</button>
`;
